File: app/src/main/java/com/e/hardviews/MyActionsDBDao.java
package com.e.hardviews;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

@Dao
public interface MyActionsDBDao {
    @Query("SELECT * FROM `Action`")
    List<Action> getAll();

    @Query("SELECT * FROM `Action` WHERE id = :id")
    Action getById(int id);

    @Insert
    void insert(List<Action> actionsList);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(Action action);

    @Update
    void update(Action action);

    @Delete
    void delete(Action action);

    @Query("DELETE FROM `Action`")
    void deleteAll();
}

File: app/src/main/java/com/e/hardviews/MyActionsDBDao.kt
package com.e.hardviews

import androidx.lifecycle.LiveData
import androidx.room.*

@Dao
interface MyActionsDBDao {
    @get:Query("SELECT * FROM `Action`")
    val all: LiveData<List<Action>>

    @Query("SELECT * FROM `Action` WHERE id = :id")
    fun getById(id: Int): Action?

    @Insert
    fun insert(actionsList: List<Action?>?)

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insert(action: Action?)

    @Update
    fun update(action: Action?)

    @Delete
    fun delete(action: Action?)

    @Query("DELETE FROM `Action`")
    fun deleteAll()
}